torch/csrc/jit/passes/memory_planning.cpp

```diff
@@ -0,0 +1,399 @@
+#include <torch/csrc/jit/passes/memory_planning.h>
+
+#include <jit/tensorexpr/kernel.h>
+#include <torch/csrc/jit/ir/alias_analysis.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/runtime/static/ops.h>
+#include <limits>
+
+namespace torch {
+namespace jit {
+
+int overlap(size_t a, size_t b, size_t c, size_t d) {
```

Comment:
You are not returning a bool result because `overlap` is used for both LiveRange and MemRegion. Is it possible to unify both by making LiveRange in the `[,)` form?

Reply:
yea you could by adding 1 to the right endpoint since [a,b] doesn't intersect [c,d] if [a,b+1) doesn't intersect [c,d+1) (since all endpoints are integers). i guess that's simpler